src/cmd-ore-wrapper

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+# NOTE: PYTHONUNBUFFERED is set in the entrypoint for unbuffered output
+#
+# This is a generic wrapper around 'ore' commands and serves
+# as a replacement for 'cmd-*-replicate' and 'cmd-buildextend-*'.
+
+import logging as log
+import sys
+
+from cosalib.cli import (
+    cloud_clis,
+    get_cloud_cli,
+    get_cloud_ore_cmds,
+    BuildCli
+)
+from cosalib.qemuvariants import get_qemu_variant
+
+if __name__ == '__main__':
+    log.basicConfig(
+        format='[%(asctime)s  %(levelname)s]: %(message)s',
+        level=log.INFO)
+
+    parser = BuildCli(
+        description="""interface for running commands against ore
+Each target has its own sub options. To access them us:
+    '--target <target> --help'
+""",
+        add_help=False,
+    )
+
+    # Determine if this a symlink, default to that.
+    sys_target = None
+    replicate = False
+    build_build = False
+
+    # Check if this is a legacy interface
+    if "-replicate" in sys.argv[0]:
+        log.info("symlink is for a replication command")
+        replicate = True
+
+    if "cmd-buildextend" in sys.argv[0]:
+        log.info("symlink is for a build and publish command")
+        build_build = True
+
+    # previous cmd-buildextend-<target> used symlinks
+    for k in cloud_clis():
+        if k in sys.argv[0]:
+            sys_target = k
+            log.info(f"ore target {sys_target} found via symlink")
+            break
+    parser.add_argument("--target",
+                        default=sys_target,
+                        choices=cloud_clis(),
+                        help="Target type for ore command")
+    pre_args = parser.parse_known_args()[0]
+    target = pre_args.target
+    log.debug(f"extending cli for {target}")
```

Comment:
Not sure what's happening here; what do these three lines actually do?

Reply:
That's some dark arts:
- we process the known args
- since the target is a bare word, we get the first argument which should be the $TARGET
- then we call the target's cli extendor.